app/controllers/search_controller.rb

```diff
@@ -121,4 +133,76 @@ def format_record_type(record_type, obsolete = false)
     record_text
   end
 
-end
+  def group_results_by_ontology(results)
```

Comment:
if you don't re-use this function in multiple places, just write it in where used